DBの設計(テーブルの追加)/README.md

senthgit7th opened the pull request:
#What
chat-spaceの実装にあたり、基礎となるデータベースの設計準備を行う目的。

1/ groups_usersテーブル（グループとユーザーの中間テーブル）
・ユーザーID　・グループID
2/usersテーブル（ユーザーの登録に関するテーブル）
・名前(ニックネーム)　・登録されるメールアドレス
3/groupsテーブル（チャットグループに関するテーブル）
・グループ名
4/messagesテーブル（チャットの投稿に関するテーブル）
・投稿テキスト　・投稿画像　・投稿者名　・グループ名

#Why
実装に入る前段階準備として、機能の洗い出しと必要となるデータベースを確認・作成するため。

senthgit7th committed:
Update README.md

senthgit7th committed:
Update README.md

senthgit7th committed:
Update README.md

groupsテーブルの追加

senthgit7th committed:
Update README.md

messageテーブルの追加

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+### Association
+- has_many :groups_users
+- has_many :groups,through: groups_users
+- has_many :massages
```
messagesですね。スペルご確認お願い致します。

senthgit7th replied:
@shingoteshima
スペルミス失礼致しました。修正致しました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false, unique: true|
```
nameで結構です。

senthgit7th replied:
@shingoteshima
group_name　→　name　修正致しました。

shingoteshima commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
```
外部キーをつける際はintegerではなくreferences型にしましょう。またその際は`_id`は不要です。
他の箇所も同様です。

senthgit7th replied:
@shingoteshima
外部キー該当箇所のタイプ修正、idの削除致しました。

senthgit7th committed:
Update README.md

has_many :massages
→messages

|group_name|string|null: false, unique: true|
→|name|string|null: false, unique: true|

外部キー該当箇所
→referencesに変更

i-date commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, add_index: true|
```
Optionで指定するので、`add_index: true`ではなくて`index: true`にしましょう。

senthgit7th replied:
@i-date
修正致しました。

i-date commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+
+### Association
+- has_many :groups_users
+- has_many :groups,through: groups_users
```
細かいですが、46行目後半の`through: groups_users`に`:`を一つ足して`through: :groups_users`としましょう。

senthgit7th replied:
@i-date
修正致しました。こちらの点に関しまして、カリキュラム内で見落とししておりました。
ありがとうございます。

i-date commented on README.md:
```diff
@@ -22,3 +22,50 @@ Things you may want to cover:
+
+### Association
+- has_many :groups_users
+- has_many :users,through: groups_users
```
同様に57行目後半の`through: groups_users`に`:`を一つ足して`through: :groups_users`としましょう。

senthgit7th replied:
@i-date
修正致しました。こちらの点に関しまして、カリキュラム内で見落とししておりました。
ありがとうございます。

i-date reviewed (commented):
追加コメントいたしました^^

senthgit7th committed:
Update README.md

|name|string|null: false, add_index: true|
→|name|string|null: false, index: true|

through: group_users
→through: :group_users

shingoteshima commented:
LGTMです！
